context/context.go

```diff
@@ -45,6 +46,12 @@ func ScopeString(s int) string {
 	}
 }
 
+var fastlySubRegex = regexp.MustCompile("^vcl_(recv|hash|hit|miss|pass|fetch|error|deliver|log)$")
+
+func IsFastlySubroutine(name string) bool {
+	return fastlySubRegex.MatchString(name)
+}
+
```

Comment:
The regular expression in Go sometimes slow (frequently talked about this 🙃 ) and Fastly's reserved subroutines do not have too many patterns.

How about using a defined map and search from it?

```suggestion
var fastlyReservedSubroutines = map[string]struct{}{
	"vcl_recv": {},
	"vcl_hash": {},
	"vcl_hit": {},
	"vcl_miss": {},
	"vcl_pass": {},
	"vcl_fetch": {},
	"vcl_error": {},
	"vcl_deliver": {},
	"vcl_log": {},
} 

func IsFastlySubroutine(name string) bool {
	_, ok :=  fastlyReservedSubroutines[name]
	return ok
}
```

But this is a tiny thing and your change looks neat ✨   it's up to you 👍

Reply:
Thanks! Pushed another commit which replaces regexp with `map[string]bool`.